src/dbnode/storage/stats/query_stats.go

```diff
@@ -37,17 +37,33 @@ type queryStats struct {
 type noOpQueryStats struct {
 }
 
+var (
+	_ QueryStats = (*queryStats)(nil)
+	_ QueryStats = (*noOpQueryStats)(nil)
+)
+
 // QueryStats provides an interface for updating query stats.
 type QueryStats interface {
 	Update(newDocs int) error
 	Start()
 	Stop()
 }
 
+// QueryStatsValues stores values of query stats.
+type QueryStatsValues struct {
+	RecentDocs int64
+	NewDocs    int
```

Comment:
nit: maybe make this `int64` as well?

Reply:
So, I actually believe "RecentDocs" isn't really relevant at all since this is a global stats tracker but "RecentDocs" is scoped to per query.

Can we separate the API perhaps to:
```go
QueryAddNewDocs(n int64) error
QueryCompletedDocs(n int64) error
```

Ideally we would somehow call the second API `QueryCompletedDocs(...)` once a query has finished matching all potential blocks. We probably also want to use a histogram rather than a gauge to measure the calls to that API. That way we can get an idea of how big the incoming queries are (histogram of how many blocks are being matched on a per query basis over time).

That does mean we need to track that number across all blocks queried and emit just once at the end of "queryWithSpan" in "src/dbnode/storage/index.go".

Thoughts @linasm ?

Reply:
@robskillington a histogram of queries / number of blocks per query sounds really useful. But I'd rather add that in a separate PR as a new feature (since this one PR is intended to fix an existing bug).

Reply:
@rallen090 the value of `NewDocs` comes from `len(batch)` which is an `int`.

Reply:
Yeah I agree w/ moving forward with this fix for now.

And yeah the `len(batch)` is an `int` but we ultimately cast multiple times when we actually use it, so I was thinking we cast immediately instead and then the types are consistent between `NewDocs` and `RecentDocs`. But again not a big difference

Reply:
Changed `NewDocs` to `int64`.